src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunSingleAssemblyCompilationModuleGroup.cs

```diff
@@ -209,7 +210,12 @@ public override bool GeneratesPInvoke(MethodDesc method)
             if (((EcmaMethod)method).Module.Equals(method.Context.SystemModule))
                 return true;
 
-            return !Marshaller.IsMarshallingRequired(method);
+            Marshaller.IsMarshallingRequired(method, out bool marshallingRequired, out bool hasByrefArgs);
+
+            // We can generate a PInvoke ILStub if no marshalling is required. If there are byref arguments,
+            // we can still generate and precompile the stub: we just cannot inline the pinvoke transition into
```

Comment:
Is making it possible to pregenerate the stub for methods taking byrefs worth the additional complexities this is introducing? There are other marshallers that don't really do much we could allow here on the same grounds as blittable byref parameters (e.g. bool marshallers are also "easy" - we could pregenerate them and consider them "not marshalling").

We decided pinning or transforming is considered marshalling no matter how easy it is so that the system doesn't become too complicated to understand/explain.

If the bug you're addressing can be fixed by doing a foreach over the signature of the method in `pInvokeMarshalingRequired` and additionally returning true if any of the parameters is a byref, I would just do that.